### PR TITLE
Fix headers passed into HttpAsyncHook

### LINKED ISSUE #32390: Fix HttpAsyncHook headers
### Body

The hook uses `_headers`:

https://github.com/apache/airflow/blob/9276310a43d17a9e9e38c2cb83686a15656896b2/airflow/providers/http/hooks/http.py#L340-L341

but passes `headers` to the async function

https://github.com/apache/airflow/blob/9276310a43d17a9e9e38c2cb83686a15656896b2/airflow/providers/http/hooks/http.py#L368-L375

The task:

`headers=headers` needs to be `headers=_headers`

https://github.com/apache/airflow/blob/9276310a43d17a9e9e38c2cb83686a15656896b2/airflow/providers/http/hooks/http.py#L372

There was attempt to address it in https://github.com/apache/airflow/pull/31010 but the PR become stale as no response from the author.

### Committer

- [X] I acknowledge that I am a maintainer/committer of the Apache Airflow project.

### PR DESCRIPTION
**Pull request by sumeshpremraj**
This fixes the header value passed into the HttpAsyncHook and adds a test, and uses a backport of unittest.mock for AsyncMock usage in Python 3.7.

closes: #32390
replaces: https://github.com/apache/airflow/pull/31010

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -118,6 +118,7 @@ install_requires =
     markupsafe>=1.1.1
     marshmallow-oneofschema>=2.0.1
     mdit-py-plugins>=0.3.0
+    mock>=5.0.2;python_version<="3.7"
```
Comment (eladkal) [inline setup.cfg]: We dont support Python 3.7 any longer.
If you check the test suite you will see it tests against 3.8 - 3.11

Where did you notice failure related to Python 3.7?

Comment (sumeshpremraj) [reply setup.cfg]: In the stale PR https://github.com/apache/airflow/actions/runs/4885483415/jobs/8719703717?pr=31010
I understand Py 3.7 support will be dropped only in the next Airflow release, but I didn't realize the test suites have already dropped it, sorry about that.

I've reverted the Py 3.7 related changes.

Comment (eladkal) [reply setup.cfg]: This is provider code :)
You can see in the change log that we already dropped support
https://airflow.apache.org/docs/apache-airflow-providers-github/stable/index.html#id1

Python 3.7 is relevant only for Airflow 2.6 releases

**Review by eladkal — APPROVED** [review]

File: setup.cfg (hunk 1 above)
Comment (sumeshpremraj) [reply setup.cfg]: Ah TIL, this is my first time contributing to Airflow, I've only been a user until now.
Thanks for the review 😄

**Review by potiuk — APPROVED** [review]

**Pull request merged**